bibxml/xml2rfc_adapters.py

```diff
@@ -503,6 +503,23 @@ def resolve_docid(self) -> Optional[DocID]:
         return None
 
 
+    def resolve(self) -> BibliographicItem:
+        refs = self.fetch_refs()
+        if num_refs := len(refs):
+            self.log(f"{num_refs} found")
+            resolved_item = self.build_bibitem_from_refs(refs)
+            link = as_list(resolved_item.link or [])
+            for index, _ in enumerate(link):
+                parsed_link = urlparse(link[index].content)
+                if parsed_link.scheme == "http":
+                    link[index].content = \
+                        parsed_link._replace(scheme="https").geturl()
```

Comment:
I wonder if there's a public method for this (i.e. `_replace`). Please ignore otherwise.

Reply:
Python documentation is using urllib documentation says no? https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse